mongo/src/main/java/com/redhat/lightblue/mongo/crud/MongoSafeUpdateProtocol.java

```diff
@@ -201,20 +235,20 @@ public int getSize() {
      */
     public Map<Integer,Error> commit() {
         Map<Integer,Error> results=new HashMap<>();
-        if(!idsInBatch.isEmpty()) {
+        if(!batch.isEmpty()) {
             BulkWriteResult writeResult;
-            LOGGER.debug("attemptToUpdate={}",idsInBatch.size());
+            LOGGER.debug("attemptToUpdate={}",batch.size());
             try {
                 if(writeConcern==null) {
                     writeResult=bwo.execute();
                 } else {
                     writeResult=bwo.execute(writeConcern);
                 }
                 LOGGER.debug("writeResult={}",writeResult);
-                if(idsInBatch.size()==writeResult.getMatchedCount()) {
+                if(batch.size()==writeResult.getMatchedCount()) {
                     LOGGER.debug("Successful update");
                 } else {
-                    LOGGER.debug("notUpdated={}",idsInBatch.size()-writeResult.getMatchedCount());
+                    LOGGER.debug("notUpdated={}",batch.size()-writeResult.getMatchedCount());
```

Comment:
Perhaps we should log this as WARN? It would be good to know how often it happens. Knowing that could help us understand the performance impact if we decide to take this further (run the query again to include new documents in the retry).